File: CS/MainForm.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrinting.Native;
using DevExpress.XtraReports.UI;

namespace dxSample {
    public partial class MainForm : DevExpress.XtraEditors.XtraForm {
        public MainForm() {
            InitializeComponent();

            XtraReport rep = new XtraReportMain();
            rep.CreateDocument();
            documentViewer1.DocumentSource = rep;
        }
        public float GetTopIndent() {
            float sumHeight = 0;
            XtraReportMain mainReport = documentViewer1.DocumentSource as XtraReportMain;
            NestedBrickIterator iterator = new NestedBrickIterator(mainReport.Pages[mainReport.Pages.Count - 1].InnerBricks);
            while (iterator.MoveNext()) {
                if (iterator.CurrentBrick is VisualBrick) {
                    VisualBrick brick = (VisualBrick)iterator.CurrentBrick;
                    float bottomPos = brick.Rect.Bottom;

                    if (bottomPos > sumHeight) {
                        sumHeight = bottomPos;
                    }
                }
            }
            sumHeight = GraphicsUnitConverter.Convert(sumHeight, GraphicsUnit.Document, GraphicsUnit.Inch) * 100;
            return sumHeight;
        }
        private void GenerateSecondReport() {
            XtraReport report = CreateReport();
            new ReportPrintTool(report).ShowPreview();
        }
        private XtraReport CreateReport() {
            float topIndent = GetTopIndent();
            XtraReport report = new XtraReport1();
            ReportHeaderBand reportHeaderBand = InitReportHeaderBand(topIndent, report);
            report.Bands.Add(reportHeaderBand);
            report.CreateDocument();
            return report;
        }
        private static ReportHeaderBand InitReportHeaderBand(float topIndent, XtraReport report) {
            ReportHeaderBand reportHeaderBand = report.Bands.OfType<ReportHeaderBand>().FirstOrDefault();
            XRPanel indentPanel = new XRPanel() {
                HeightF = topIndent, LocationF = new PointF(0, 0)
            };
            if(reportHeaderBand == null) {
                reportHeaderBand = new ReportHeaderBand();
                reportHeaderBand.Controls.Add(indentPanel);
            } else {
                reportHeaderBand.Controls.Add(indentPanel);
                reportHeaderBand.Controls.SetChildIndex(indentPanel, 0);
                reportHeaderBand.Controls[1].LocationF = new PointF(indentPanel.LocationF.X, indentPanel.BottomF);
            }
            return reportHeaderBand;
        }
        private void barButtonItem3_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            GenerateSecondReport();
        }
    }
}

File: VB/MainForm.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrinting.Native
Imports DevExpress.XtraReports.UI

Namespace dxSample
    Partial Public Class MainForm
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()

            Dim rep As XtraReport = New XtraReportMain()
            rep.CreateDocument()
            documentViewer1.DocumentSource = rep
        End Sub
        Public Function GetTopIndent() As Single
            Dim sumHeight As Single = 0
            Dim mainReport As XtraReportMain = TryCast(documentViewer1.DocumentSource, XtraReportMain)
            Dim [iterator] As New NestedBrickIterator(mainReport.Pages(mainReport.Pages.Count - 1).InnerBricks)
            Do While [iterator].MoveNext()
                If TypeOf [iterator].CurrentBrick Is VisualBrick Then
                    Dim brick As VisualBrick = CType([iterator].CurrentBrick, VisualBrick)
                    Dim bottomPos As Single = brick.Rect.Bottom

                    If bottomPos > sumHeight Then
                        sumHeight = bottomPos
                    End If
                End If
            Loop
            sumHeight = GraphicsUnitConverter.Convert(sumHeight, GraphicsUnit.Document, GraphicsUnit.Inch) * 100
            Return sumHeight
        End Function
        Private Sub GenerateSecondReport()
            Dim report As XtraReport = CreateReport()
            CType(New ReportPrintTool(report), ReportPrintTool).ShowPreview()
        End Sub
        Private Function CreateReport() As XtraReport
            Dim topIndent As Single = GetTopIndent()
            Dim report As XtraReport = New XtraReport1()
            Dim reportHeaderBand As ReportHeaderBand = InitReportHeaderBand(topIndent, report)
            report.Bands.Add(reportHeaderBand)
            report.CreateDocument()
            Return report
        End Function
        Private Shared Function InitReportHeaderBand(ByVal topIndent As Single, ByVal report As XtraReport) As ReportHeaderBand
            Dim reportHeaderBand As ReportHeaderBand = report.Bands.OfType(Of ReportHeaderBand)().FirstOrDefault()
            Dim indentPanel As New XRPanel() With {.HeightF = topIndent, .LocationF = New PointF(0, 0)}
            If reportHeaderBand Is Nothing Then
                reportHeaderBand = New ReportHeaderBand()
                reportHeaderBand.Controls.Add(indentPanel)
            Else
                reportHeaderBand.Controls.Add(indentPanel)
                reportHeaderBand.Controls.SetChildIndex(indentPanel, 0)
                reportHeaderBand.Controls(1).LocationF = New PointF(indentPanel.LocationF.X, indentPanel.BottomF)
            End If
            Return reportHeaderBand
        End Function
        Private Sub barButtonItem3_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonItem3.ItemClick
            GenerateSecondReport()
        End Sub
    End Class
End Namespace
